core/txpool/txpool.go

```diff
@@ -190,13 +198,16 @@ func (p *TxPool) loop(head *types.Header, chain BlockChain) {
 	var (
 		resetBusy = make(chan struct{}, 1) // Allow 1 reset to run concurrently
 		resetDone = make(chan *types.Header)
+
+		resetForced bool       // Whether a forced reset was requested, only used in simulator mode
+		resetWaiter chan error // Channel waitinf on a forced reset, only used in simulator mode
```

Comment:
Nitpick, theoretically if we receive a `SYNC` request and then the txpool is closed afterwards, the SYNC waiter will be blocked forever. Should we add a defer to handle it?

```go
	// Notify the live reset waiter to not block forever if the txpool is closed.
	defer func() {
		if resetWaiter != nil {
			resetWaiter <- nil
			resetWaiter = nil
		}
	}()
```